src/Microsoft.IdentityModel.Tokens/Encryption/AuthenticatedEncryptionProvider.cs

```diff
@@ -165,13 +180,19 @@ private AuthenticatedEncryptionResult EncryptWithAesCbc(byte[] plaintext, byte[]
         private byte[] DecryptWithAesCbc(byte[] ciphertext, byte[] authenticatedData, byte[] iv, byte[] authenticationTag)
         {
             // Verify authentication Tag
+            int expectedTagLength = _authenticatedkeys.Value.HmacKey.Key.Length;
+            if (ShouldValidateAuthenticationTagLength()
+                && ExpectedAuthenticationTagBase64UrlSizeInBytes.TryGetValue(Algorithm, out expectedTagLength)
+                && expectedTagLength != authenticationTag.Length)
+                throw LogHelper.LogExceptionMessage(new SecurityTokenDecryptionFailedException(LogHelper.FormatInvariant(LogMessages.IDX10625, Base64UrlEncoder.Encode(authenticationTag))));
+
             byte[] al = Utility.ConvertToBigEndian(authenticatedData.Length * 8);
             byte[] macBytes = new byte[authenticatedData.Length + iv.Length + ciphertext.Length + al.Length];
             Array.Copy(authenticatedData, 0, macBytes, 0, authenticatedData.Length);
             Array.Copy(iv, 0, macBytes, authenticatedData.Length, iv.Length);
             Array.Copy(ciphertext, 0, macBytes, authenticatedData.Length + iv.Length, ciphertext.Length);
             Array.Copy(al, 0, macBytes, authenticatedData.Length + iv.Length + ciphertext.Length, al.Length);
-            if (!_symmetricSignatureProvider.Value.Verify(macBytes, 0, macBytes.Length, authenticationTag, 0, _authenticatedkeys.Value.HmacKey.Key.Length, Algorithm))
```

Comment:
@brentschmaltz we discussed possibly replacing this length - this would remove the necessity for the check on lines 183-188. However, then it wouldn't be behind the app context switch. I propose that we keep it as is (_authenticatedkeys.Value.HmacKey.Key.Length) and update it upon the next major release since we'll remove the context switch at that point.

Reply:
I went ahead and kept it "_authenticatedkeys.Value.HmacKey.Key.Length" for the same reason why we want an app context switch. We can update this on the next major rel of Wilson